src/java.base/share/classes/java/io/ObjectInputFilter.java

```diff
@@ -581,11 +592,13 @@ static ObjectInputFilter rejectUndecidedClass(ObjectInputFilter filter) {
         private static volatile BinaryOperator<ObjectInputFilter> serialFilterFactory;
 
         /**
-         * Boolean to indicate that the filter factory has been used by an ObjectInputStream
-         * constructor and cannot be changed thereafter.
+         * Boolean to indicate that the filter factory can not be set or replaced.
+         * - an ObjectInputStream has already been created using the current filter factory
+         * - has been set on the command line
+         * - jdk.serialFilter is set and jdk.serialFilterFactory is unset, the builtin can not be replaced
          * @see Config#setSerialFilterFactory(BinaryOperator)
          */
-        private static final AtomicBoolean filterFactoryUsed = new AtomicBoolean(false);
+        private static final AtomicBoolean filterFactoryNoReplace = new AtomicBoolean(false);
```

Comment:
Nit: This could simply be `new AtomicBoolean()`; IIRC it saves a volatile write.